src/server/v2.0/handler/retention.go

```diff
@@ -0,0 +1,354 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/goharbor/harbor/src/common/rbac"
+	projectCtl "github.com/goharbor/harbor/src/controller/project"
+	retentionCtl "github.com/goharbor/harbor/src/controller/retention"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/project/metadata"
+	"github.com/goharbor/harbor/src/pkg/retention/policy"
+	"github.com/goharbor/harbor/src/pkg/task"
+	"github.com/goharbor/harbor/src/server/v2.0/handler/model"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/retention"
+)
+
+func newRetentionAPI() *retentionAPI {
+	return &retentionAPI{
+		projectCtl:   projectCtl.Ctl,
+		retentionCtl: retentionCtl.Ctl,
+		proMetaMgr:   metadata.Mgr,
+	}
+}
+
+// RetentionAPI ...
+type retentionAPI struct {
+	BaseAPI
+	proMetaMgr   metadata.Manager
+	retentionCtl retentionCtl.APIController
+	projectCtl   projectCtl.Controller
+}
+
+func (r *retentionAPI) Prepare(ctx context.Context, operation string, params interface{}) middleware.Responder {
+	if err := r.RequireAuthenticated(ctx); err != nil {
+		return r.SendError(ctx, err)
+	}
+
+	return nil
+}
+
+func (r *retentionAPI) GetRentenitionMetadata(ctx context.Context, params operation.GetRentenitionMetadataParams) middleware.Responder {
+	payload := &models.RetentionMetadata{
```

Comment:
Move this out of this method and define it as var in the package to reduce the memory allocated.